plugins/network-elements/nuage-vsp/src/com/cloud/network/manager/NuageVspManagerImpl.java

```diff
@@ -152,53 +226,181 @@ public NuageVspDeviceVO addNuageVspDevice(AddNuageVspDeviceCmd cmd) {
+                }
+                return nuageVspDevice;
             } else {
                 throw new CloudRuntimeException("Failed to add Nuage Vsp Device due to internal error.");
             }
-        } catch (ConfigurationException e) {
+        } catch (Exception e) {
             throw new CloudRuntimeException(e.getMessage());
         }
     }
 
     @Override
+    public NuageVspDeviceVO updateNuageVspDevice(UpdateNuageVspDeviceCmd command) {
+        ServerResource resource = new NuageVspResource();
+        final String deviceName = Network.Provider.NuageVsp.getName();
+        ExternalNetworkDeviceManager.NetworkDevice networkDevice = ExternalNetworkDeviceManager.NetworkDevice.getNetworkDevice(deviceName);
+        final Long physicalNetworkId = command.getPhysicalNetworkId();
+        PhysicalNetworkVO physicalNetwork = _physicalNetworkDao.findById(physicalNetworkId);
+        if (physicalNetwork == null) {
+            throw new InvalidParameterValueException("Could not find phyical network with ID: " + physicalNetworkId);
+        }
+
+        final PhysicalNetworkServiceProviderVO ntwkSvcProvider = _physicalNetworkServiceProviderDao.findByServiceProvider(physicalNetwork.getId(),
+                networkDevice.getNetworkServiceProvder());
+        if (ntwkSvcProvider == null) {
+            throw new CloudRuntimeException("Network Service Provider: " + networkDevice.getNetworkServiceProvder() + " is not enabled in the physical network: "
+                    + physicalNetworkId + "to add this device");
+        } else if (ntwkSvcProvider.getState() == PhysicalNetworkServiceProvider.State.Shutdown) {
+            throw new CloudRuntimeException("Network Service Provider: " + ntwkSvcProvider.getProviderName() + " is in shutdown state in the physical network: "
+                    + physicalNetworkId + "to add this device");
+        }
+
+        HostVO nuageVspHost = null;
+        NuageVspDeviceVO nuageVspDevice = null;
+        List<NuageVspDeviceVO> nuageVspDevices = _nuageVspDao.listByPhysicalNetwork(physicalNetworkId);
+        if (nuageVspDevices == null || nuageVspDevices.size() == 0) {
+            throw new CloudRuntimeException("Nuage VSD is not configured on physical network " + physicalNetworkId);
+        } else {
+            nuageVspDevice = nuageVspDevices.iterator().next();
+            nuageVspHost = _hostDao.findById(nuageVspDevice.getHostId());
+            _hostDao.loadDetails(nuageVspHost);
+        }
+
+        Map<String, String> paramsTobeUpdated = new HashMap<String, String>();
+        if (StringUtils.isNotBlank(command.getHostName()) &&
+                !command.getHostName().equals(nuageVspHost.getDetails().get("hostname"))) {
+            paramsTobeUpdated.put("name", "Nuage VSD - " + command.getHostName());
+            paramsTobeUpdated.put("hostname", command.getHostName());
+        }
+
+        if (StringUtils.isNotBlank(command.getUserName()) &&
```

Comment:
Use Guava's implementation rather Apache Commons.  There is no need for the duplication.